infrastructure/src/main/java/org/corfudb/infrastructure/log/Compression/ZSTDCompression.java

```diff
@@ -0,0 +1,67 @@
+package org.corfudb.infrastructure.log.Compression;
+
+import com.github.luben.zstd.Zstd;
+
+import java.nio.ByteBuffer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class ZSTDCompression implements Codec {
+
+    private static final int DEFAULT_COMPRESSION_LEVEL = 3;
+
+    private static ZSTDCompression INSTANCE = new ZSTDCompression();
+
+    public static ZSTDCompression getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public ByteBuffer compress(ByteBuffer uncompressed) {
+        checkNotNull(uncompressed);
+        checkArgument(uncompressed.hasRemaining());
+
+        final int decompressedLength = uncompressed.remaining();
+        final int maxCompressedLength = (int) Zstd.compressBound(decompressedLength);
```

Comment:
why not use long instead of int?

Reply:
Because its used to allocate an array which requires an int. The cast can be moved to where the allocation happens L:29